chronos_bin/src/message_receiver.rs

```diff
@@ -18,69 +17,88 @@ pub struct MessageReceiver {
 }
 
 impl MessageReceiver {
-    // pub fn new(consumer: Arc<KafkaConsumer>, producer: Arc<KafkaProducer>, data_store: Arc<Pg>) -> Self {
-    //     Self {
-    //         consumer,
-    //         producer,
-    //         data_store,
-    //     }
-    // }
     pub async fn run(&self) {
-        println!("Receiver ON!");
+        log::info!("Receiver ON!");
         let _ = &self.consumer.subscribe().await;
         // for _n in 0..100 {
         let mut total_count = 0;
         let mut direct_sent_count = 0;
         let mut db_insert_count = 0;
         loop {
+            log::debug!("Receiver loop");
             if let Ok(message) = &self.consumer.consume_message().await {
                 total_count += 1;
-                if headers_check(message.headers().unwrap()) {
-                    let new_message = &message;
-                    let headers = required_headers(new_message).expect("parsing headers failed");
-                    let message_deadline: DateTime<Utc> = DateTime::<Utc>::from_str(&headers[DEADLINE]).expect("String date parsing failed");
+                if let Some(headers) = message.headers() {
+                    if headers_check(headers) {
+                        let new_message = &message;
+                        if let Some(headers) = required_headers(new_message) {
+                            if let Ok(message_deadline) = DateTime::<Utc>::from_str(&headers[DEADLINE]) {
+                                if message_deadline <= Utc::now() {
+                                    direct_sent_count += 1;
+                                    let string_payload = match get_payload_utf8(new_message) {
+                                        Some(string_payload) => String::from_utf8_lossy(string_payload).to_string(),
+                                        None => {
+                                            log::error!("message payload is not utf8");
```

Comment:
When debugging this in production you almost always want to see the original value which we failed to parse. Add `new_message` to this log line.